test/gtest/uct/ib/test_cqe_zipping.cc

```diff
@@ -0,0 +1,161 @@
+                                     UCT_IB_IFACE_STAT_RX_COMPLETION_ZIPPED};
+        size_t zipped_cqes_count  = 0;
+
+        for (auto &entity : m_entities) {
+            auto stats = ucs_derived_of(entity->iface(), uct_ib_iface_t)->stats;
+            for (auto &counter : counters) {
+                zipped_cqes_count += UCS_STATS_GET_COUNTER(stats, counter);
+            }
+        }
+        return zipped_cqes_count;
+#else
+        return 0;
+#endif
+    }
+
+    virtual void init()
+    {
+#if defined(ENABLE_STATS)
+        stats_activate();
+#endif
+        modify_config("IB_CQE_ZIPPING_ENABLE", "y");
+
+        test_uct_ib_with_specific_port::init();
+
+        create_connected_entities();
+        uct_iface_set_am_handler(receiver()->iface(), 0, am_cb, &m_recv_cnt, 0);
+        m_send_buf = new mapped_buffer(m_buf_size, 0, *sender());
+    }
+
+    void flush()
+    {
+        uct_test::flush();
+        m_send_cnt = m_recv_cnt = 0;
+    }
+
+private:
+    entity *sender() const
+    {
+        return m_e1;
+    }
+
+    entity *receiver() const
+    {
+        return m_e2;
+    }
+
+    virtual void cleanup()
+    {
+        delete m_send_buf;
+        uct_test::cleanup();
+
+#if defined(ENABLE_STATS)
+        stats_restore();
+#endif
+    }
+
+    ucs_status_t am_zcopy()
+    {
+        auto size = sender()->iface_attr().cap.am.max_zcopy;
+        size      = ucs_min(size, m_buf_size);
+        uct_iov_t iov{m_send_buf->ptr(), size, m_send_buf->memh(), 0, 1};
+
+        return uct_ep_am_zcopy(sender()->ep(0), 0, m_send_buf->ptr(), 0, &iov,
+                               1, 0, nullptr);
+    }
+
+    static ucs_status_t
+    am_cb(void *arg, void *data, size_t length, unsigned flags)
+    {
+        size_t &recv_cnt = *(static_cast<size_t*>(arg));
+        ++recv_cnt;
+        return UCS_OK;
+    }
+
+    void check_port_attr()
+    {
+        if (IBV_PORT_IS_LINK_LAYER_ETHERNET(&m_port_attr)) {
+            /* Hard to get zipped CQEs on Ethernet link */
+            UCS_TEST_SKIP_R("skip CQE zipping test for Ethernet link type");
+        }
+    }
+
+    static constexpr size_t m_buf_size = 2048;
+    size_t                  m_send_cnt = 0, m_recv_cnt = 0;
+    mapped_buffer           *m_send_buf{nullptr};
+};
+
+/* We test only ZCOPY with 4K size messages due to good PCI load and stably
+ * CQE zipping reproducing on all platforms.
+ */
+UCS_TEST_P(test_cqe_zipping, zcopy)
+{
+    int deadline_seconds      = check_counters() ? 90 : 5;
+    const ucs_time_t deadline = ucs::get_deadline(deadline_seconds);
+
+    while ((ucs_get_time() < deadline) && (get_total_zipped_count() == 0)) {
+        /* Flush resets the transort resources which were acquired during the
+         * previous iteration.
+         */
+        flush();
+        send_while_possible();
+        wait_for_completion();
+    }
+
+    ASSERT_TRUE((!check_counters()) || (get_total_zipped_count() > 0));
```

Comment:
```
if (is_cqe_zipping_expected()) {
  EXPECT_GE(get_total_zipped_count(), 0);
}
```

Reply:
The overall proposal is good, but I think we should use `EXPECT_GT` here, don't we?

Reply:
yes, need EXPECT_GT